mount/mount_darwin.go

```diff
@@ -0,0 +1,85 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package mount
+
+import (
+	"fmt"
+	"os/exec"
+
+	"github.com/containerd/containerd/errdefs"
+	"github.com/containerd/containerd/log"
+	"github.com/pkg/errors"
+)
+
+var (
+	// ErrNotImplementOnUnix is returned for methods that are not implemented
+	ErrNotImplementOnUnix = errors.New("not implemented under darwin")
+	// allowedHelperBinaries is a list of third-party executable, which can
+	// be extended in future uses, e.g., mount_nfs, mount_9p.  `mount_darwin`
+	// is a special case where we will use the internal darwin snapshotter.
+	allowedHelperBinaries = []string{"mount_darwin"}
```

Comment:
1. Not, in this PR.
2. Yes, but the point is not really about extensibility. The purpose of using helper binaries is to split the controversial part to a non-official repo, so that we can soon reach consensus on merging this PR. We can revisit merging the repos later at any time.

Reply:
this part (decouple to `mount_containerd_*`) is already available here (https://github.com/ukontainer/darwin-snapshotter).  I believe this comment is addressed.